dynamodb/src/main/scala/awscala/dynamodbv2/Table.scala

```diff
@@ -51,6 +53,25 @@ case class Table(
   def put(hashPK: Any, attributes: (String, Any)*)(implicit dynamoDB: DynamoDB): Unit = putItem(hashPK, attributes: _*)
   def put(hashPK: Any, rangePK: Any, attributes: (String, Any)*)(implicit dynamoDB: DynamoDB): Unit = putItem(hashPK, rangePK, attributes: _*)
 
+  def put[E <: AnyRef](hashPK: Any, entity: E)(implicit dynamoDB: DynamoDB, c: ClassTag[E]) = {
+    dynamoDB.put(this, hashPK, getAttrValues(entity): _*)
+  }
+
+  def put[E <: AnyRef](hashPK: Any, rangePK: AnyRef, entity: E)(implicit dynamoDB: DynamoDB,c: ClassTag[E]) = {
+    dynamoDB.put(this, hashPK, rangePK, getAttrValues(entity): _*)
+  }
+
+  def getAttrValues(entity: AnyRef) = {
+    val fields = entity.getClass.getDeclaredFields.map(_.getName).toList
+    fields.map(f => f -> getValueFromEntity(f, entity))
+  }
+
+  def getValueFromEntity(field: String, entity: AnyRef) = {
+    val fv = entity.getClass.getDeclaredField(field)
```

Comment:
Instead of directly access private fields, we should use only public getter methods.

Reply:
Refactored this part of code to make it more readable. Thanks!